Content.Client/Research/UI/ResearchConsoleMenu.xaml.cs

```diff
@@ -60,10 +66,14 @@ public void  UpdatePanels(ResearchConsoleBoundInterfaceState state)
         {
             MinHeight = 10
         });
+
+        var hasAccess = _player.LocalPlayer?.ControlledEntity is not { } local ||
+                        !_entity.TryGetComponent<AccessReaderComponent>(Entity, out var access) ||
+                        _accessReader.IsAllowed(local, access);
```

Comment:
I think making it a field in the state, then only doing one check in the server would be better

Reply:
The only issue i had with that is that whether or not an individual has access to the server can vary across clients. The ui state updating isn't bound to a specific person looking at it so this ended up being the cleanest way to do it. Plus, it shouldn't be a huge problem considering access is predicted and this is only getting checked once.